cmd/kubectl-gadget/main.go

```diff
@@ -78,7 +80,23 @@ func main() {
 		// evaluate flags early for runtimeGlobalFlags; this will make
 		// sure that --connection-method has already been parsed when calling
 		// InitDeployInfo()
-		rootCmd.ParseFlags(os.Args[1:])
+
+		// Do not error out on unknown flags, but still validate currently
+		// known ones.
+		// Other flags will be validated in the `Execute()` call and unknown
+		// ones will be rejected
+		rootCmd.FParseErrWhitelist.UnknownFlags = true
+		// temporary workaround for https://github.com/inspektor-gadget/inspektor-gadget/pull/2174#issuecomment-1780923952
+		args := commonutils.RemoveSplitSortArgs(os.Args[1:])
+		// --help should be handled after we registered all commands
+		args = commonutils.RemoveHelpArg(args)
+		err := rootCmd.ParseFlags(args)
+		if err != nil {
+			// Analogous to cobra error message
+			fmt.Fprintf(os.Stderr, "Error: %v\n", err)
+			os.Exit(1)
+		}
+		rootCmd.FParseErrWhitelist.UnknownFlags = false
```

Comment:
Perhaps all of it could go to a helper in commonutils and be reused from kubectl-gadget and gadgetctl?

Something like: 

```golang
// ParseEarlyFlags evaluates flags that are needed to connect to the server.
func ParseEarlyFlags(cmd *cobra.Cmd, args []string) error
```